google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb

```diff
@@ -422,14 +423,16 @@ def dead_letter_topic= new_dead_letter_topic
         # acknowledgement deadline has been exceeded for the message). A NACK is any call to ModifyAckDeadline with a 0
         # deadline. Note that client libraries may automatically extend ack_deadlines.
         #
-        # This field will be honored on a best effort basis. If this parameter is 0, a default value of 5 is used.
+        # This field will be honored on a best effort basis. If this parameter is `nil` or `0`, a default value of `5`
+        # is used.
         #
-        # See also {#dead_letter_max_delivery_attempts=}, {#dead_letter_topic=} and {#dead_letter_topic}.
+        # See also {#dead_letter_max_delivery_attempts=}, {#dead_letter_topic=}, {#dead_letter_topic},
+        # {#dead_letter_policy?} and {#remove_dead_letter_policy!}.
         #
         # Makes an API call to retrieve the value when called on a reference object. See {#reference?}.
         #
-        # @return [Integer, nil] A value between 5 and 100, or `nil` if no dead letter policy is configured. If this
-        #   value is 0, a default value of 5 is used.
+        # @return [Integer, nil] A value between `5` and `100`, or `nil` if no dead letter policy is configured. If this
+        #   value was set to `0` or `nil`, a default value of `5` is used.
```

Comment:
Seems to be some confusion about what `nil` means: whether it is because no dead letter policy is configured, or a default value is requested.

Reply:
Removed the second sentence.